packages/core/src/routes/hook.ts

```diff
@@ -23,11 +26,28 @@ export default function hookRoutes<T extends AuthedRouter>(
 
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: Hooks.createGuard
+        .omit({ id: true, event: true, enabled: true })
+        .extend({ config: hookConfigGuard.omit({ signingKey: true }) }),
+      response: Hooks.guard,
+      status: [200, 422],
+    }),
     async (ctx, next) => {
+      const { config, events, ...rest } = ctx.guard.body;
+      assertThat(events.length > 0, new RequestError({ code: 'hook.empty_events', status: 422 }));
```

Comment:
nit: should we add min(1) guard on the events ? Currently, if the events field is undefined invalid body error will be thrown, but if it is an empty array hook.empty_events is thrown instead.

Reply:
Excellent! Add an `.nonempty()` to ensure the `events` have at least one element.
https://zod.dev/?id=nonempty